src/player.js

```diff
@@ -2,22 +2,21 @@ import VideoPlayer from './video-player';
 import { defaultProfiles } from './config/profiles';
 import { isRawUrl } from './plugins/cloudinary/common';
 
-export const getProfile = async (cloudName, profile) => {
+export const getProfile = async (cloudName, profile, secureDistribution) => {
   if (Object.keys(defaultProfiles).includes(profile)) {
     return defaultProfiles[profile];
   }
 
-  if (isRawUrl(profile)) {
-    return await fetch(profile, { method: 'GET' }).then(res => res.json());
-  }
-
-  throw new Error('Custom profiles will be supported soon, please use one of default profiles: "cldDefault", "cldLooping" or "cldAdaptiveStream"');
+  const cloudinaryDomain = secureDistribution === 'res-staging.cloudinary.com' ? secureDistribution : 'res.cloudinary.com';
+  const profileUrl = isRawUrl(profile) ? profile :
+    `https://${cloudinaryDomain}/${cloudName}/_applet_/video_service/video_player_profiles/${profile}.json`;
```

Comment:
There are many considerations to take when building Cloudinary URLs (`private_cdn`, `cname` etc), you should use the `url-gen` SDK to handle that, we already import `createCloudinaryLegacyURL` for video URLs

https://github.com/cloudinary/js-url-gen/blob/master/src/backwards/createCloudinaryLegacyURL.ts